skimage/color/colorconv.py

```diff
@@ -789,10 +808,22 @@ def rgb2gray(rgb):
     >>> img_gray = rgb2gray(img)
     """
 
-    if rgb.ndim == 2:
+    rgb = np.asanyarray(rgb)
+
+    if rgb.ndim == 2 and rgb.shape[-1] != 3:
```

Comment:
I would really like to see a PR with **only** these changes.

Reply:
I will open such a PR ;-)

Reply:
Please see #4438